src/matchers/cognito.internal.ts

```diff
@@ -0,0 +1,188 @@
+        : `Expected "${userPoolId}" user pool to have a user with username "${username}"`,
+    pass: pass,
+  };
+});
+
+export const toExistAndMatchObject = withRetry(async function (
+  this: MatcherContext,
+  input: CognitoUserInput,
+  expected: AdminGetUserResponse,
+) {
+  const matcherName = 'toExistAndMatchObject';
+  const options: MatcherHintOptions = {
+    isNot: this.isNot,
+  };
+
+  const { userPoolId, username, clientConfig } = input;
+
+  const client = getCognitoClient(clientConfig);
+
+  try {
+    const { $metadata, ...received } = await client.send(
+      new AdminGetUserCommand({
+        UserPoolId: userPoolId,
+        Username: username,
+      }),
+    );
+
+    const pass = equals(received, expected, [iterableEquality, subsetEquality]);
+
+    const message = pass
+      ? () =>
+          matcherHint(matcherName, undefined, undefined, options) +
+          '\n\n' +
+          `Expected: not ${printExpected(expected)}\n` +
+          (expected !== received
+            ? `Received:     ${printReceived(received)}`
+            : '')
+      : () =>
+          matcherHint(matcherName, undefined, undefined, options) +
+          '\n\n' +
+          printDiffOrStringify(
+            expected,
+            received,
+            EXPECTED_LABEL,
+            RECEIVED_LABEL,
+            this.expand !== false,
+          );
+
+    return { actual: received, expected, message, name: matcherName, pass };
+  } catch (e) {
+    if (e instanceof UserNotFoundException) {
+      return {
+        message: () =>
+          `Expected "${userPoolId}" user pool to have a user with username "${username}"`,
+        pass: false,
+      };
+    }
+
+    throw e;
+  }
+});
+
+export const toExistAndMatchSnapshot = withRetry(async function (
+  this: Context,
+  input: CognitoUserInput,
+  ...rest: any
+) {
+  const { userPoolId, username, clientConfig } = input;
+
+  const client = getCognitoClient(clientConfig);
+  try {
+    const { $metadata, ...received } = await client.send(
+      new AdminGetUserCommand({
+        UserPoolId: userPoolId,
+        Username: username,
+      }),
+    );
+
+    if (!received) {
+      return {
+        message: () =>
+          `Expected "${userPoolId}" user pool to have user with username "${username}"`,
+        pass: false,
+      };
+    }
+
+    return toMatchSnapshot.call(this, received, ...rest);
+  } catch (e) {
+    if (e instanceof UserNotFoundException) {
+      return {
+        message: () =>
+          `Expected "${userPoolId}" user pool to have a user with username "${username}"`,
+        pass: false,
+      };
+    }
+
+    throw e;
+  }
+});
+
+export const toExistAndMatchInlineSnapshot = withRetry(async function (
+  this: Context,
+  input: CognitoUserInput,
+  ...rest: any
+) {
+  const { userPoolId, username, clientConfig } = input;
+
+  const client = getCognitoClient(clientConfig);
+
+  try {
+    const { $metadata, ...received } = await client.send(
```

Comment:
```suggestion
   const { $metadata, ...received } = await client.send(
```

and add 
```json
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
        "varsIgnorePattern": "^_"
      }
    ]
```

in `.eslintrc`